libraries/app/api.cpp

```diff
@@ -73,6 +77,87 @@ namespace graphene { namespace app {
        return true;
     }
 
+    bool login_api::login_signed( const string& b64_encoded_trx )
+    {
+      const string trx_json = fc::base64_decode( b64_encoded_trx );
+      auto var = fc::json::from_string( trx_json );
+      auto trx = var.as<signed_transaction>( 200 );
+
+      auto db = _app.chain_database();
+      int64_t offset( (trx.expiration - db->head_block_time()).to_seconds() );
+      if( offset > 5*60 || offset < 0 )
+         return false; 
+
+      auto op = trx.operations[0];
+      if( op.which() != operation::tag<transfer_operation>::value ) // only transfer op for validation
```

Comment:
Can we please add a check that `from==`to`? That way the tx becomes invalid for broadcast.

Reply:
The trx isn't broadcasted at all. 

It's just the operation which is used to check for a correct signature. I am not 100% sure, but normally you could fill the trx data with anything you like. 

I could not set an `amount` and only set me as `from` and no `to` and the signature should still be valid.

Reply:
I know it's not broadcast, but if it was valid, it **could** be broadcast. I would like to prevent that from happening (think MITM attack)

Reply:
Ah yeah you are right. Will sure do that.